### PR TITLE
Add _unstableHotReload to chunk mapper.

### PR DESCRIPTION
**Pull request by Hyperkid123**
Prerequisite for hot loading for module development.

Hot reload requires only a single runtime chunk which breaks our chunk mapper (no entry chunk is found).

Added new _**Unstable**_ way to look for entry chunks:

```JSON
{
  "starter": {
    "entry": [
      "/beta/apps/starter/starter.js",
      "/beta/apps/starter/js/runtime.js"
    ]
  }
}
```

Entry runtime chunks are now two, one for federated modules and one for global runtime. The federated module entry chunk is a part of the runtime chunk and requires it in order to initialize the webpack sharing scope.

These two should be now separate to not break existing builds but there is a possibility that we could use the same method for finding chunks in bot scenarios.

### CODE REVIEW

File: packages/config-utils/chunk-mapper.js [hunk 1]
```diff
@@ -1,35 +1,59 @@
 class ChunkMapper {
   constructor(options) {
+    const { _unstableHotReload, ...rest } = options;
     this.config = {};
-    this.options = options || {};
+    this.options = rest || {};
+    this._unstableHotReload = _unstableHotReload;
+    this.hotChunkRegex = RegExp('\\.hot-update\\.js$');
```
Comment (karelhala) [inline packages/config-utils/chunk-mapper.js:7]: Can we add an option to customize this? By default we'd use the hot-update regexp, but we could add option to customize it.

**Review by karelhala — APPROVED** [review]

File: packages/config-utils/chunk-mapper.js (hunk 1 above)
Comment (Hyperkid123) [reply packages/config-utils/chunk-mapper.js:7]: not sure if there is anything to customize :thinking: . Webpack will always call the hot-update chunks the same.

Comment (karelhala) [reply packages/config-utils/chunk-mapper.js:7]: If that's the thing I don't think we have to initialize the value on constructor level. But rather have it outside constructor.

Comment (Hyperkid123) [reply packages/config-utils/chunk-mapper.js]: Done

Comment (karelhala) [reply packages/config-utils/chunk-mapper.js]: I meant more like a class member, but this works as well :smiley:

Comment (Hyperkid123) [reply packages/config-utils/chunk-mapper.js]: Yeah sorry :D it does the same thing anyways

**Pull request merged**